java/writer.go

```diff
@@ -0,0 +1,59 @@
+package java
+
+import (
+	"strings"
+)
+
+type CodeWriter interface {
+	Begin()
+	End()
+	Eol()
+	Write(code string)
+}
+
+func prefix(indentation int) string {
+	tab := strings.Repeat(" ", 2)
+	return strings.Repeat(tab, indentation)
+}
+
+type codeWriter struct {
+	code        strings.Builder
+	indentation int
+	newLine     bool
+}
+
+func (self *codeWriter) Begin() {
+	self.Write(" {")
+	self.Eol()
+	self.indentation += 1
+}
+
+func (self *codeWriter) End() {
+	self.indentation -= 1
+	self.Write("}")
+	self.Eol()
+}
+
+func (self *codeWriter) Eol() {
+	if self.newLine {
+		self.newLine = false
```

Comment:
Scala and I'm guessing Typescript need this fix to prevent empty lines from having spaces at the beginning.

Reply:
This occurs when you have a double newline? Or some other scenario?

Reply:
@chris-smith-zocdoc yea double newline